src/multi.jl

```diff
@@ -1,35 +1,95 @@
 
     # Variable holders
     λ = Dict()  # Extreme points and multipliers
     α = Dict()  # Partitioning Variables
     β = Dict()  # Lifted variables for exact formulation
 
-    # Construct λ variable space
-    for k in keys(m.nonlinear_terms)
-        nl_type = m.nonlinear_terms[k][:nonlinear_type]
-        if ((nl_type == :multilinear) || (nl_type == :bilinear)) && (m.nonlinear_terms[k][:convexified] == false)
-            λ, α = amp_convexify_multilinear(m, k, λ, α, discretization)
-        elseif nl_type == :monomial && !m.nonlinear_terms[k][:convexified]
-            λ, α = amp_convexify_monomial(m, k, λ, α, discretization)
-        elseif nl_type == :binlin && !m.nonlinear_terms[k][:convexified]
-            amp_convexify_binlin(m, k, discretization)
-        elseif nl_type == :binprod && !m.nonlinear_terms[k][:convexified]
+    # Convexification Treatment for Complex Non-Convex Terms
+    for k in keys(m.nonconvex_terms)
+        nl_type = m.nonconvex_terms[k][:nonlinear_type]
+        if ((nl_type == :MULTILINEAR) || (nl_type == :BILINEAR)) && (m.nonconvex_terms[k][:convexified] == false)
+            λ, α = amp_convexify_multilinear(m, k, λ, α, d)
+        elseif nl_type == :MONOMIAL && !m.nonconvex_terms[k][:convexified]
+            λ, α = amp_convexify_monomial(m, k, λ, α, d)
+        elseif nl_type == :BINLIN && !m.nonconvex_terms[k][:convexified]
+            β = amp_convexify_binlin(m, k, β)
+        elseif nl_type == :BINPROD && !m.nonconvex_terms[k][:convexified]
             β = amp_convexify_binprod(m, k, β)
+        elseif nl_type == :BININT && !m.nonconvex_terms[k][:convexified]
+            β = amp_convexify_binint(m, k, β)
+        elseif nl_type == :INTPROD && !m.nonconvex_terms[k][:convexified]
+            m.int_enable || error("Integer features are OFF. No support for INTPROD at this condition.")
+            λ, α = amp_convexify_intprod(m, k, λ, α, d)
+        elseif nl_type == :INTLIN && !m.nonconvex_terms[k][:convexified]
+            m.int_enable || error("Integer features are OFF. No support for INTLIN at this condition.")
+            λ, α = amp_convexify_intlin(m, k, λ, α, d)
+        elseif nl_type in [:sin, :cos] && !m.nonconvex_terms[k][:convexified]
+            λ, α = amp_convexify_sincos(m, k, λ, α, d)
         end
     end
 
+    # Experimental Code for Integer Problems
+    if m.int_enable
+        for i in 1:m.num_var_orig
+            m.var_type[i] == :Int && amp_convexify_integer(m, i, λ, α, d)
+        end
+    end
+
+    # Experimental code for Warm starting
+    m.convhull_warmstart && !isempty(m.best_bound_sol) && amp_warmstart_α(m, α)
+
     return
 end
 
-function amp_convexify_multilinear(m::PODNonlinearModel, k, λ::Dict, α::Dict, discretization::Dict)
-    m.nonlinear_terms[k][:convexified] = true  # Bookeeping the convexified terms
+function amp_convexify_integer(m::PODNonlinearModel, idx::Int, λ::Dict, α::Dict, d::Dict)
```

Comment:
What does this function do?

Reply:
This is an experimental function that is used to handle integer problems (beyond 0/1). I added a docstring to this function in the most recent commit.